### PR TITLE
refactor broker plugins with new tracing base plugin

### PR DESCRIPTION
**Pull request by rochdev**
### What does this PR do?

Refactor broker plugins with new tracing base plugin.

### Motivation

Continuation of #2373

### CODE REVIEW

File: packages/dd-trace/test/plugins/helpers.js
Comment (tlhunter) [inline packages/dd-trace/test/plugins/helpers.js:13]: What's our stance on commented out code?

Comment (rochdev) [reply packages/dd-trace/test/plugins/helpers.js:13]: Stance is they should be removed, which I will do right now.

File: packages/datadog-plugin-google-cloud-pubsub/src/producer.js
Comment (tlhunter) [inline packages/datadog-plugin-google-cloud-pubsub/src/producer.js:13]: Are these TODOs before merging or in the future?

File: packages/datadog-plugin-graphql/src/execute.js
Comment (tlhunter) [inline packages/datadog-plugin-graphql/src/execute.js:66]: What kind of errors would be caught in here?

File: packages/datadog-plugin-rhea/src/consumer.js
Comment (tlhunter) [inline packages/datadog-plugin-rhea/src/consumer.js:12]: Should this be removed now?

**Review by tlhunter — COMMENTED** [review]
Seems mostly good but I'm curious about the TODOs.

It's a pretty large refactor so I'm not sure if I'm yet qualified to approve ;)

File: packages/datadog-plugin-rhea/src/consumer.js
Comment (rochdev) [reply packages/datadog-plugin-rhea/src/consumer.js:12]: I guess this could have been better worded, but it's a breaking change so it can only be removed in a major. Will update to reflect that.

Comment (rochdev) [reply packages/datadog-plugin-rhea/src/consumer.js:12]: Actually I'm just gonna remove the TODO as I already put in the 4.0 plan to review AMQP integrations.

File: packages/datadog-plugin-graphql/src/execute.js
Comment (rochdev) [reply packages/datadog-plugin-graphql/src/execute.js:66]: If the document is invalid, the tool is unavailable, or the signature cannot be generated for any reason, then we should just fallback to returning the original document source instead of throwing.

File: packages/datadog-plugin-google-cloud-pubsub/src/producer.js
Comment (rochdev) [reply packages/datadog-plugin-google-cloud-pubsub/src/producer.js:13]: In the future. For producers and consumers, the span name is supposed to end with `.consume` and `.produce` which was not the case historically. This is a breaking change so it can only be done in a major.

**Review by Qard — APPROVED** [review]
LGTM but just some small questions.

File: packages/datadog-plugin-rhea/src/consumer.js
Comment (rochdev) [reply packages/datadog-plugin-rhea/src/consumer.js:38]: `amq.topic` is the default exchange name in [RabbitMQ](https://www.rabbitmq.com/tutorials/amqp-concepts.html#exchanges), and the name was carried over for AMQP 1.0 integrations.

File: packages/datadog-plugin-amqplib/test/index.spec.js
Comment (rochdev) [reply packages/datadog-plugin-amqplib/test/index.spec.js:174]: These tags are meant to report the outgoing host when making a request as a client or a producer. For the consumer, a request is received, so even though we might be able to capture this information, it's not the intended purpose of these tags. I can add them back and schedule them for deprecation instead if you prefer.

Comment (Qard) [reply packages/datadog-plugin-amqplib/test/index.spec.js:174]: Nope, I'm good calling it a bug fix if it was not the intended behaviour. Just wasn't clear why it was disappearing.

File: packages/datadog-plugin-rhea/src/consumer.js
Comment (Qard) [reply packages/datadog-plugin-rhea/src/consumer.js:38]: Ah, okay. 👍🏻

File: packages/datadog-plugin-grpc/src/util.js
Comment (hguptajc) [inline packages/datadog-plugin-grpc/src/util.js:7]: @rochdev we were using v2.4.0 in our repo but since some vuln was fixed in the later version we have to upgrade to v2.46.0.
Our repo used `addMethodTags` to add tags to the span. Since it was deprecated with this pr, and the tags are just returned  in `getMethodMetadata` is the caller expected to add the tags to the span? Moreover maintain the older format of tags for example:
```{
      'grpc.method.path': path,
      'grpc.method.kind': kind
    }
```

Comment (rochdev) [reply packages/datadog-plugin-grpc/src/util.js:7]: This method was never exposed publicly, so if you were using it in your code I would say yes to adding the tags manually, which is the safe way to do it regardless of the existence of that method.

Comment (hguptajc) [reply packages/datadog-plugin-grpc/src/util.js:7]: Thank you! I understand. 
Is there a DD way to report the gRPC data? We would want to defer to that and not maintain a separate system.

Comment (rochdev) [reply packages/datadog-plugin-grpc/src/util.js:7]: Are you using the auto-instrumentation? I guess I'm not sure I understand the use case since these should be added automatically in the first place.

**Pull request merged**
